doc/srv6/SRv6_uSID.md

```diff
@@ -0,0 +1,112 @@
+# SONiC uSID
+
+## Table of Content
+- [Overview](#Overview)
+- [Scope](#Scope)
+- [Design](#Design)
+- [Example](#Example)
+
+## Revision
+
+| Rev  |   Date    |              Author              | Change Description      |
+| :--: | :-------: | :------------------------------: | :---------------------: |
+| 0.1  | 7/17/2022 | Shitanshu Shah, Reshma Sudarshan |  Initial version        |
+
+
+## Overview
+SRv6 uSID (micro-segment) is extension of the SRv6 network programming model [SRv6 uSID instructions IETF draft](https://datatracker.ietf.org/doc/draft-filsfils-spring-net-pgm-extension-srv6-usid/). uSID is a compressed SID value which can be for example carried in 16-bits (unlike full IPv6 address to represent a SID). uSID as is designed scales well with much lower MTU overhead required per uSID carrier. uSID carrier is 128-bit IPv6 address that can carry upto 6 uSIDs [Refer to Example for more details]
+
+## Scope
+The scope of this document is to enhance orchagent to support uSID programming instructions in this IETF draft. Current SAI API definitions already support uSID instructions. No SAI API change required in scope of this document. Current version of routing protocols in SONiC does not support SRv6, it is not in the scope of this document to add such a support for FRR routing stack.
+
+## Design
+Current srv6orch is designed, per [SRv6 HLD](https://github.com/sonic-net/SONiC/blob/master/doc/srv6/srv6_hld.md), to support SRv6 programming instructions as described in RFC 8754 and RFC 8986. This design extends SRv6 Network Programming with a new type of SRv6 SID behaviors defined as uSID.

```

Comment:
Is the uSID functionality natively supported in FRR or does it need to be upstreamed? what is the plan or is there any PR to be reviewed? please share if there is any PR for community to review.